src/canary-notify-slack/@canary-notify-slack.yml

```diff
@@ -0,0 +1,3 @@
+version: 2.1
+description: |
+  Send canary notifications to Slack containing custom fields.
```

Comment:
Instead of being specific to canaries, could this be a general "send build success/failure notifications to slack" orb?

I see there is https://circleci.com/orbs/registry/orb/circleci/slack but maybe it's a bit too general.